### ISSUE TITLE
False positive for cast_sign_loss

### ISSUE THREAD

**Issue by Luro02** [rust-lang/rust-clippy#4743]
## `cargo clippy -V`
```
clippy 0.0.212 (52cebb1 2019-10-15)
```
## Example code
```rust
#![warn(clippy::cast_sign_loss)]

fn main() {
    let value = i64::min_value().checked_abs().unwrap_or(0) as u64;
    
    assert_eq!(value, 0);
}
```
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=81cf39e78646285ec19d8b42f2d2a96b)

## Problem
Clippy does warn, that the sign could be lost implicitly, even though it is done explicitly with [`checked_abs`](https://doc.rust-lang.org/std/primitive.i64.html#method.checked_abs), which either returns `Some(positive i64)` or `None`.

## Expected behavior
Clippy should ignore this case.

**Comment by flip1995** [comment]
Easy extension of #4614

**Comment by krishna-veerareddy** [comment]
@Luro02 Hey this has been fixed by #4883.

**Comment by phansch** [comment]
This should be fixed with #4883
